File: balance-app/src/main/java/com/petro/balance/model/BalanceChangeEvent.java
package com.petro.balance.model;

import java.time.Instant;
import java.util.concurrent.ThreadLocalRandom;

public record BalanceChangeEvent(String name, int balance, Instant publishedAt) {

  public BalanceChangeEvent(String name) {
    this(name, ThreadLocalRandom.current().nextInt(0,1500), Instant.now());
  }

}

File: balance-app-kotlin/src/main/kotlin/com/petro/balance/model/BalanceChangeEvent.kt
package com.petro.balance.model

import java.time.Instant
import java.util.concurrent.ThreadLocalRandom

data class BalanceChangeEvent(
    val name: String,
    val balance: Int = ThreadLocalRandom.current().nextInt(0, 1500),
    val publishedAt: Instant = Instant.now(),
)
